src/cluster_legacy.c

```diff
@@ -2632,6 +2744,36 @@ uint32_t writePingExt(clusterMsg *hdr, int gossipcount) {
         extensions++;
     }
 
+    if (sdslen(myself->announce_client_ipv4) != 0) {
+        if (cursor != NULL) {
+            /* Populate announce_client_ipv4 */
+            clusterMsgPingExtClientIpV4 *ext =
+                preparePingExt(cursor, CLUSTERMSG_EXT_TYPE_CLIENT_IPv4, getClientIpV4PingExtSize());
+            memcpy(ext->announce_client_ipv4, myself->announce_client_ipv4, sdslen(myself->announce_client_ipv4));
+
+            /* Move the write cursor */
+            cursor = getNextPingExt(cursor);
+        }
+
+        totlen += getClientIpV4PingExtSize();
+        extensions++;
+    }
```

Comment:
This code block is almost identical for all 4 extensions (hostname, human_nodename, ipv4, and ipv6). Can we extract it out and create a common helper function? something like the below?